--- a/gcc/c-family/c-common.h
+++ b/gcc/c-family/c-common.h
@@ -88,6 +88,7 @@ enum rid
   RID_COMPLEX,
   RID_THREAD,
   RID_TYPEOF,
+  RID_MEM_SHARED,
   
   /* C extensions */
   RID_ALIGNAS, RID_ALIGNOF, RID_ATOMIC, RID_GENERIC, RID_IMAGINARY,
@@ -134,6 +135,9 @@ enum rid
 #define D_CXXWARNOLDSTYLE  0x004	/* warn about C style casts */
 #define D_CXXWARNCONVERSION 0x008	/* warn about type conversion issues */
 
+/* mem_shared keyword flags */
+#define D_MEM_SHARED       0x010        /* mem_shared keyword */
+
 struct c_common_resword
 {
   const char *word;